### DISCUSSION TITLE
installing cloudflare warp

### DISCUSSION THREAD

**Discussion by deltaforce** [elementary/terminal#686] [General]
Disclaimer: I am a complete linux noob, so bear this in mind when replying.

I am trying to install cloudflare warp following these instructions
https://developers.cloudflare.com/warp-client/get-started/linux/
https://pkg.cloudflareclient.com/install

But all I get after following all the steps  is "the repository does not have a release file"

**Comment by lewisgoddard** [comment]
### Installing Cloudflare WARP on elementary OS 6.x

The problem is `$(lsb_release -cs)` returning `jolnir` on elementary 6.1, and not `focal` (the Ubuntu LTS base). Changing this in the `/etc/apt/sources.list.d/cloudflare-client.list` file resolves the issue, or for a clean install run this one line at a time:

```sh
curl https://pkg.cloudflareclient.com/pubkey.gpg | sudo gpg --yes --dearmor --output /usr/share/keyrings/cloudflare-warp-archive-keyring.gpg
echo "deb [arch=amd64 signed-by=/usr/share/keyrings/cloudflare-warp-archive-keyring.gpg] https://pkg.cloudflareclient.com/ focal main" | sudo tee /etc/apt/sources.list.d/cloudflare-client.list
sudo apt update
````

### Installing Cloudflare WARP on elementary OS 7.x

Newer releases will be based on Ubuntu 22.04 LTS, so will need to swap in `jammy` instead.

**Comment by carlajarchuleta1-a11y** [comment]
curl https://pkg.cloudflareclient.com/pubkey.gpg | sudo gpg --yes --dearmor --output /usr/share/keyrings/cloudflare-warp-archive-keyring.gpg
echo "deb [arch=amd64 signed-by=/usr/share/keyrings/cloudflare-warp-archive-keyring.gpg] https://pkg.cloudflareclient.com/ focal main" | sudo tee /etc/apt/sources.list.d/cloudflare-client.list
sudo apt update
